File: PeopleManager/Teacher.cpp
#include "stdafx.h"
#include "Teacher.h"


int Teacher::nextId = 0;
Teacher::Teacher()
	: Teacher("Stukkinn D. Amudd", 45, "English")

{
	
}

Teacher::Teacher(std::string name,
	int age,
	std::string specialistSubject)
	: Person(name, age)
{
	this->specialistSubject = specialistSubject;
	teacherID = nextId;
	nextId++;
}


Teacher::~Teacher()
{
}


void Teacher::display()
{
	std::cout << "\nDisplaying information for a teacher...\n"
		<< "Name: " << getName() << "\n"
		<< "Age: " << getAge() << "\n"
		<< "Teacher ID: " << getTeacherID() << "\n"
		<< "Specialist subject: " << getSpecialistSubject() << "\n";
}


int Teacher::getTeacherID()
{
	return teacherID;
}


std::string Teacher::getSpecialistSubject()
{
	return specialistSubject;
}

File: PeopleManagerCS/Teacher.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PeopleManagerCS
{
    public class Teacher
        : Person
    {
        public Teacher()
            : this("Stik Indie Mudd", 36, "English")
        { }

        public Teacher(string name, int age, string specialistSubject)
            : base(name, age)
        {
            SpecialistSubject = specialistSubject;
        }
        public string TeacherID { get; } =
            Guid.NewGuid().ToString();
        public string SpecialistSubject { get; set; }

        public override void Display()
        {
            Console.WriteLine("Displaying information for a teacher...");
            Console.WriteLine($"Name: { Name }\nTeacher ID: { TeacherID }\nAge: { Age }\nSpecialist subject: { SpecialistSubject }");
            Console.WriteLine("Press any key to continue...");
            Console.ReadKey();
        }

    }
}
